### ISSUE TITLE
Out-of-source MSVC Debug fails to link

### ISSUE THREAD

**Issue by PBfordev** [KeyWorksRW/wxUiEditor#589]
To avoid changes in the source tree, out-of-source builds are commonly used and recommended. However, wxUiEditor's CMakefile does not support this and uses relative paths instead of relying on `CMAKE_SOURCE_DIR`.

This means that building out-of-source Debug build with MSVC will fail with "_Error LNK1181 cannot open input file '../src/wxui.natvis'_", since the path is invalid.

The issue could be fixed with this change
```diff
CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c110284..4482163 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,7 +74,7 @@ if (MSVC)
     target_link_options(wxUiEditor PRIVATE "$<$<CONFIG:Release>:/LTCG>")
 
     target_link_options(wxUiEditor PRIVATE
-        "$<$<CONFIG:Debug>:/natvis:../src/wxui.natvis>"
+        "$<$<CONFIG:Debug>:/natvis:${CMAKE_SOURCE_DIR}/src/wxui.natvis>"
         "$<$<CONFIG:Debug>:/OPT:REF>"
     )
 
```
However, doing this single change is probably not a good idea.

**Comment by KeyWorksRW** [comment]
This is the only place where a relative path was used. Everything else either assumes the path CMakeLists.txt is in, or it uses ${CMAKE_CURRENT_LIST_DIR}.

What directory were you building from? I don't want to close out the issue until I've verified building from a similar directory. The github codeql-analysis action uses MSVC but doesn't invoke the linker, so I don't have an automatic way to verify the fix.

**Comment by PBfordev** [comment]
The source was in `D:\Dev\Desktop\wxUiEditor-GIT` folder.

The build was in `D:\Dev\Desktop\wxUiEditor-GIT-build-vc16_x64` folder.

**Comment by PBfordev** [comment]
Fixed in d999055.
